src/components/BoxForTransaction.tsx

```diff
@@ -138,49 +143,93 @@ export const BoxForTransaction = () => {
         }
 
         try {
-          dispatch({
-            type: "setState",
-            payload: {
-              loadingMessage: "Processing Transaction...",
-            },
-          })
+          await web3.eth
+            .estimateGas(transactionParameters)
+            .then(function (gasEstimate) {
+              transactionParameters.gas = gasEstimate
+            })
 
-          const response = await web3.eth.sendTransaction(transactionParameters)
+          try {
+            dispatch({
```

Comment:
This `dispatch` call got moved down but should come before any `await`ing is done